### ISSUE TITLE
iOS - VoiceOver Reading French and Spanish Content in English

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#125]
For some reason now, VoiceOver is reading the Spanish and French Comply Chain screens all in English even when the text changes to the correct language.

To me, that means the screen/page language is not changing as it should to the updated language.

**Comment by Abychalil** [comment]
Hi John,
We couldn't recreate this issue. VoiceOver is reading the content in French and Spanish, when I changed the language. Please retest and advise if this issue is still there. 
@jfc3-dol

**Comment by jfc3-dol** [comment]
@Abychalil 

Not sure why but both of my iOS devices are reading the other languages with an English accent.

**Comment by Abychalil** [comment]
Hi John, Please see the comment from developer as below.  

"1) Issue #125 - Voice-over still reading French and Spanish content in "English Accent" - that is because the chosen System language is English and not really an accessibility bug. The Voiceover or Talkback is reading Spanish and French as expected and the Accent Voice is not really something a developer can control or change."

@jfc3-dol

**Comment by jfc3-dol** [comment]
@Abychalil 

The language should be set by the application to read in the language that a person chooses to hear from VoiceOver.

I understand since my default language is English, if the application does not set the language a person asks for from it, then it will default to my chosen language. 

If I were talking about this with a friend that spoke French better than English and they wanted to read stuff on my phone and choose French for the application on my phone with VoiceOver, they would expect to hear French and not French text with an English accent. The application should override the default language the person chooses to the one they want to hear from VoiceOver or a web-based screen reader or TalkBack on Andriod too.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

As we talked about the other day, you are looking into fixing the larger problem with all three platforms (iOS, Android, and web) to have the screen readers use the language set by the application, not the platform, because people might want to use a different language.

**Comment by harpreetsdev** [comment]
@jfc3-dol For this issue I feel the solution is larger than the Comply chain app since we tested some other language related features in addition to the accent. We might have to get on a phone call again to discuss this one.

The fix for this is pretty straight forward. Go to  Settings -> Accessibility -> Voice Over -> Speech -> Add New Language. That newly added language should take care of these accent issues (**In an ideal world**). But this feature has reportedly stopped working after IOS 10 OS update.

I created a small video and uploaded to Google Drive. You might have to request access -- 

https://drive.google.com/file/d/1l52tV1YRZ9cw5fWwwaG7mZutU4PHoSZ1/view?usp=sharing

A couple of apple support forum discussions on this matter- 

https://discussions.apple.com/thread/7692159#:~:text=All%20you%20need%20to%20do,It%20should%20work.

https://discussions.apple.com/thread/7692159?page=3

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

As issue #166 shows in the screenshot of the web version. The page language needs to be changed to the correct language to have it read properly by a screen reader. The text is in Spanish, but the HTML lang="en" for English still is why it is reading with an English accent.

At least for the web version, and I assume for the iOS and Android too since it's using the same main codebase.

I was not able to watch/listen to the video since I don't have permission.

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

I did a quick test were I changed the language from lang="en" to lang="es" for the Spanish text and it read text with the proper language and accent. I could show you what I mean later.

**Comment by harpreetsdev** [comment]
@jfc3-dol 
Sure. We could have a chat sometime this week maybe and hatch this out? Doesn't have to be a meeting, we could just get on a quick call.

@Abychalil @pbhatt17

**Comment by jfc3-dol** [comment]
@harpreetsdev @Abychalil @pbhatt17

Sent e-mail with times available in the next two days, since I'm busy today/

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abbychalil

I am leaving this issue one until we figure out issue #172 ( Web - Page Language Changes on Page Refresh but Text Stays in non-English Language ). I want to make sure if there is a fix, it works here too.

**Comment by jfc3-dol** [comment]
Still going to keep this issue open until the refresh web page issues ( #172 ) is figured out.

**Comment by jfc3-dol** [comment]
@pbhatt17 

The Web - Page Language Changes on Page Refresh but Text Stays in Non-English Language #172 is now closed, so I'm going to close this issue.
